### PR TITLE
[GPU] KV-cache compression support

### PR DESCRIPTION
**Pull request by sshlyapn**
### Details:
This PR enables KV-cache compression support
Currently, it supports only combinations of the following configurations:
* Data types: INT8_SYM / INT8_ASYM
* Modes: per-token (quantization of `num_heads * head_size` in a single group) / per-token-per-head (quantization of each `head_size` group for each head per token)

### Tickets:
 - *ticket-id*

### CODE REVIEW

File: src/plugins/intel_gpu/include/intel_gpu/op/dynamic_quantize.hpp [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    /// \brief Constructs an DynamicQuantize operation.
+    ///
+    /// \param data Input tensor with data
+    /// \param config Dynamic quantization configuration
+    /// \param scales_zp_output_order Specifies on default order of scales and zero points
+    /// \param combine_scales_and_zp If true, combines scales and zero points into a single buffer, pairing each scale with its corresponding zero point
+    DynamicQuantize(const Output<Node>& data,
+                    const QuantizationConfig& config,
+                    const std::vector<uint64_t>& scales_zp_output_order = {},
+                    const bool combine_scales_and_zp = false);
```
Comment (vladimir-paramuzov) [inline src/plugins/intel_gpu/include/intel_gpu/op/dynamic_quantize.hpp:30]: I'd suggest adding this interleaved/planar mode for scales and zero points as a part of common operation

**Comment by ilya-lavrenov** [comment]
does it support SDPA case or PA as well?

**Comment by sshlyapn** [comment]
> does it support SDPA case or PA as well?

@ilya-lavrenov as of now only SDPA supports the compression (currently, scales/zp are stored in state variables memory in independent memory buffer). For PagedAttention, we can probably reuse some logic, but we have to consider storing all data in a single buffer (KV-cache itself and related scales and zero points)

File: src/plugins/intel_gpu/include/intel_gpu/op/dynamic_quantize.hpp (hunk 1 above)
Comment (sshlyapn) [reply src/plugins/intel_gpu/include/intel_gpu/op/dynamic_quantize.hpp]: I have updated the common DQ operation with an additional property with the following configuration:
```
    /**
     * @brief Configuration for how Activations, Scales and Zero Points will be stored in output buffers:
     * - Planar: Activations, Scales, and Zero Points are stored in independent buffers.
     * - InterleavedScalesZP: Activations are stored in an independent buffer, while Scales and Zero Points (if any) are combined in a separate buffer.
     */
    enum class OutputStorageType { Planar, InterleavedScalesZP, /* InterleavedActivationsScalesZP */ };
```
Custom GPU operation has been removed

**Review by vladimir-paramuzov — APPROVED** [review]

**Pull request merged**